### PR TITLE
SDL2 UI

### PR DESCRIPTION
**Pull request by helenmedrano**

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -17,7 +19,45 @@ lazy_static! {
...
+            match event {
+                sdl2::event::Event::Quit {..} |
+                sdl2::event::Event::KeyDown { keycode: Some(sdl2::keyboard::Keycode::Escape), .. } => {
+                    return;
+                },
+                _ => {}
+            }
+        }
+        canvas.present();
+        ::std::thread::sleep(std::time::Duration::new(0, 1_000_000_000u32 / 60));
```
Comment (wmedrano) [inline src/main.rs:56]: You are using `Duration::new` which takes seconds and nanoseconds, but never use the seconds.
https://doc.rust-lang.org/std/time/struct.Duration.html#method.new

How about using `Duration::from_nanos`
https://doc.rust-lang.org/std/time/struct.Duration.html#method.from_nanos

File: src/main.rs [hunk 2]
```diff
@@ -17,7 +19,45 @@ lazy_static! {
...
+
+    let mut canvas = window.into_canvas().build().unwrap();
+
+    canvas.set_draw_color(Color::RGB(0, 255, 255));
+    canvas.clear();
+    canvas.present();
+    let mut event_pump = sdl_context.event_pump().unwrap();
+    let mut i = 0;
+    loop {
+        i = (i+1) % 255;
```
Comment (wmedrano) [inline src/main.rs:43]: Optional: Remove the color cycling and leave a clean background.

**Comment by wmedrano** [comment]
Also make sure CI passes. You probably just need to add sdl2 to the dependencies in `.github/workflows/rust.yml`.
You can find ubuntu packages at https://packages.ubuntu.com/

File: src/main.rs (hunk 2 above)
Comment (helenmedrano) [reply src/main.rs:43]: but how else will i know it's working

**Comment by helenmedrano** [comment]
> Also make sure CI passes. You probably just need to add sdl2 to the dependencies in `.github/workflows/rust.yml`.
> You can find ubuntu packages at https://packages.ubuntu.com/

This is not a directory in the project....

**Comment by helenmedrano** [comment]
Updated color: 
Screen Shot 2020-12-03 at 9 42 38 AM

File: src/main.rs (hunk 2 above)
Comment (helenmedrano) [reply src/main.rs:43]: updated

**Comment by helenmedrano** [comment]
@wmedrano 
Do you need to push up the file:
Screen Shot 2020-12-03 at 9 44 10 AM

**Comment by wmedrano** [comment]
> @wmedrano
> Do you need to push up the file:
> Screen Shot 2020-12-03 at 9 44 10 AM

It takes it from the branch.

**Comment by wmedrano** [comment]
:taco:

**Review by wmedrano — APPROVED** [review]

**Pull request merged**